pkg/resource/v1/certconfig/desired.go

```diff
@@ -2,8 +2,375 @@ package certconfig
+		desiredCertConfigs = append(desiredCertConfigs, certConfig)
+	}
+	{
+		certConfig := newEtcdCertConfig(clusterConfig, certs.EtcdCert, r.projectName)
+		desiredCertConfigs = append(desiredCertConfigs, certConfig)
+	}
+	{
+		certConfig := newNodeOperatorCertConfig(clusterConfig, certs.NodeOperatorCert, r.projectName)
+		desiredCertConfigs = append(desiredCertConfigs, certConfig)
+	}
+	{
+		certConfig := newPrometheusCertConfig(clusterConfig, certs.PrometheusCert, r.projectName)
+		desiredCertConfigs = append(desiredCertConfigs, certConfig)
+	}
+	{
+		certConfig := newServiceAccountCertConfig(clusterConfig, certs.ServiceAccountCert, r.projectName)
+		desiredCertConfigs = append(desiredCertConfigs, certConfig)
+	}
+	{
+		certConfig := newWorkerCertConfig(clusterConfig, certs.WorkerCert, r.projectName)
+		desiredCertConfigs = append(desiredCertConfigs, certConfig)
+	}
+
+	return desiredCertConfigs, nil
+}
+
+func prepareClusterConfig(baseClusterConfig cluster.Config, clusterGuestConfig v1alpha1.ClusterGuestConfig) (*cluster.Config, error) {
+	var err error
+
+	// Copy baseClusterConfig as basis and supplement it with information from
+	// clusterGuestConfig.
+	clusterConfig := new(cluster.Config)
+	*clusterConfig = baseClusterConfig
+
+	clusterConfig.ClusterID = key.ClusterID(clusterGuestConfig)
+
+	clusterConfig.Domain.API, err = newServerDomain(key.DNSZone(clusterGuestConfig), certs.APICert)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	clusterConfig.Domain.Calico, err = newServerDomain(key.DNSZone(clusterGuestConfig), certs.CalicoCert)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	clusterConfig.Domain.Etcd, err = newServerDomain(key.DNSZone(clusterGuestConfig), certs.EtcdCert)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	clusterConfig.Domain.NodeOperator, err = newServerDomain(key.DNSZone(clusterGuestConfig), certs.NodeOperatorCert)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	clusterConfig.Domain.Prometheus, err = newServerDomain(key.DNSZone(clusterGuestConfig), certs.PrometheusCert)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	clusterConfig.Domain.ServiceAccount, err = newServerDomain(key.DNSZone(clusterGuestConfig), certs.ServiceAccountCert)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	clusterConfig.Domain.Worker, err = newServerDomain(key.DNSZone(clusterGuestConfig), certs.WorkerCert)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	clusterConfig.Organization = clusterGuestConfig.Owner
+
+	versionBundle, err := versionbundle.GetBundleByName(key.VersionBundles(clusterGuestConfig), certOperatorID)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	clusterConfig.VersionBundleVersion = versionBundle.Version
+
+	return clusterConfig, nil
+}
+
+func newAPICertConfig(clusterConfig *cluster.Config, cert certs.Cert, projectName string) *v1alpha1.CertConfig {
+	certName := string(cert)
```

Comment:
Unrelated to the PR, we should maybe implement `Stringer` for `certs.Cert`. This separated casting everywhere is not necessary then.